File: lokerpc/codegen/testdata/docs.json.ts
import { RPCContextClient } from "@loke/http-rpc-client";
import { Context } from "@loke/context";

/**
 * GetUserRequest is the request type for GetUser.
 */
export type GetUserRequest = {
  /** The unique identifier of the user. */
  id: string;
};

/**
 * User represents a user account.
 */
export type User = {
  id: string;
  /** The display name of the user. */
  name: string;
  /** Email address. Optional. */
  email?: string;
};

/**
 * A service for testing doc comments
 */
export class DocsTestService extends RPCContextClient {
  constructor(baseUrl: string) {
    super(baseUrl, "docs-test")
  }
  /**
   * Get a user by ID
   */
  getUser(ctx: Context, req: GetUserRequest): Promise<User> {
    return this.request(ctx, "getUser", req);
  }
}

File: lokerpc/codegen/testdata/docs.json.go
package docstest

import (
	"context"

	"github.com/LOKE/pkg/lokerpc"
)

// GetUserRequest is the request type for GetUser.
type GetUserRequest struct {
	// The unique identifier of the user.
	ID string `json:"id"`
}

// User represents a user account.
type User struct {
	ID string `json:"id"`
	// The display name of the user.
	Name string `json:"name"`
	// Email address. Optional.
	Email string `json:"email,omitempty"`
}

// A service for testing doc comments
type DocsTestService interface {
	// Get a user by ID
	GetUser(context.Context, GetUserRequest) (*User, error)
}

// A service for testing doc comments
type DocsTestRPCClient struct {
	lokerpc.Client
}

func (c DocsTestRPCClient) GetUser(ctx context.Context, req GetUserRequest) (*User, error) {
	var res User
	err := c.DoRequest(ctx, "getUser", req, &res)
	if err != nil {
		return nil, err
	}
	return &res, nil
}
